sonar-dotnet-shared-library/src/main/java/org/sonar/plugins/dotnet/tests/UnitTestResultsImportSensor.java

```diff
@@ -76,37 +76,40 @@ public void execute(SensorContext context) {
   }
 
   void analyze(SensorContext context, UnitTestResults unitTestResults) {
-    UnitTestResults aggregatedResults = unitTestResultsAggregator.aggregate(wildcardPatternFileProvider, unitTestResults);
+    try {
+      UnitTestResults aggregatedResults = unitTestResultsAggregator.aggregate(wildcardPatternFileProvider, unitTestResults);
 
-    context.<Integer>newMeasure()
-      .forMetric(CoreMetrics.TESTS)
-      .on(context.module())
-      .withValue(aggregatedResults.tests())
-      .save();
-    context.<Integer>newMeasure()
-      .forMetric(CoreMetrics.TEST_ERRORS)
-      .on(context.module())
-      .withValue(aggregatedResults.errors())
-      .save();
-    context.<Integer>newMeasure()
-      .forMetric(CoreMetrics.TEST_FAILURES)
-      .on(context.module())
-      .withValue(aggregatedResults.failures())
-      .save();
-    context.<Integer>newMeasure()
-      .forMetric(CoreMetrics.SKIPPED_TESTS)
-      .on(context.module())
-      .withValue(aggregatedResults.skipped())
-      .save();
-
-    Long executionTime = aggregatedResults.executionTime();
-    if (executionTime != null) {
-      context.<Long>newMeasure()
-        .forMetric(CoreMetrics.TEST_EXECUTION_TIME)
-        .on(context.module())
-        .withValue(executionTime)
+      context.<Integer>newMeasure()
+        .forMetric(CoreMetrics.TESTS)
+        .on(context.project())
+        .withValue(aggregatedResults.tests())
+        .save();
+      context.<Integer>newMeasure()
+        .forMetric(CoreMetrics.TEST_ERRORS)
+        .on(context.project())
+        .withValue(aggregatedResults.errors())
+        .save();
+      context.<Integer>newMeasure()
+        .forMetric(CoreMetrics.TEST_FAILURES)
+        .on(context.project())
+        .withValue(aggregatedResults.failures())
         .save();
+      context.<Integer>newMeasure()
+        .forMetric(CoreMetrics.SKIPPED_TESTS)
+        .on(context.project())
+        .withValue(aggregatedResults.skipped())
+        .save();
+
+      Long executionTime = aggregatedResults.executionTime();
+      if (executionTime != null) {
+        context.<Long>newMeasure()
+          .forMetric(CoreMetrics.TEST_EXECUTION_TIME)
+          .on(context.project())
+          .withValue(executionTime)
+          .save();
+      }
+    } catch (Exception e) {
+      LOG.warn("Could not import unit test report: '{}'", e.getMessage());
```

Comment:
To avoid confusion and be clear to the user, I would also add this warning message to the SonarQube UI warnings. (Using the `AnalysisWarnings` interface). So users will know right away that something did not work 100%